packages/ragbits-document-search/tests/unit/test_document_search.py

```diff
@@ -163,3 +165,40 @@ async def test_document_search_ingest_multiple_from_sources():
 
     assert len(results) == 2
     assert {result.content for result in results} == {"foo", "bar"}  # type: ignore
+
+
+async def test_document_search_with_batched():
+    documents = [
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's brother is George"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's father is Daddy Pig"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's mother is Mummy Pig"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's friend is Suzy Sheep"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's friend is Danny Dog"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's friend is Pedro Pony"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's friend is Emily Elephant"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's friend is Candy Cat"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's teacher is Madame Gazelle"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's doctor is Dr. Brown Bear"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's cousin is Chloe Pig"),
+        DocumentMeta.create_text_document_from_literal("Name of Peppa's cousin is Alexander Pig"),
+    ]
+
+    embeddings_mock = AsyncMock()
+    embeddings_mock.embed_text.return_value = [[0.1, 0.1]] * len(documents)
+
+    # processing_strategy = BatchedAsyncProcessing(batch_size=5)
```

Comment:
I think it should be used here? Rather than sequential?

Reply:
That's exactly right. Thank you!